### PR TITLE
Created clean architecture structure

### PR DESCRIPTION
**Pull request by Faka-GLB**
Created the app, data, di and domain modules. Moved main activity to activity package.

project structure:
Rick and morty modules

### CODE REVIEW

**Review by BQuinteros — APPROVED** [review]

**Review by mauroLonghi — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

File: app/build.gradle [hunk 1]
```diff
@@ -6,6 +6,10 @@ plugins {
 android {
     compileSdk 30
 
+    buildFeatures {
+        viewBinding = true
+    }
+
     defaultConfig {
         applicationId "com.onboarding.rick_and_morty_app"
         minSdk 26
```
Comment (nicolbidone) [inline app/build.gradle]: minSdk should be 23

File: data/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+/build
```
Comment (nicolbidone) [inline data/.gitignore]: missing empty line at end

File: data/build.gradle [hunk 3]
```diff
@@ -0,0 +1,44 @@
+plugins {
+    id 'com.android.library'
+    id 'kotlin-android'
+}
+
+android {
+    compileSdk 30
+
+    defaultConfig {
+        minSdk 26
```
Comment (nicolbidone) [inline data/build.gradle]: minSdk should be 23

File: data/proguard-rules.pro [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+#   public *;
+#}
+
+# Uncomment this to preserve the line number information for
+# debugging stack traces.
+#-keepattributes SourceFile,LineNumberTable
+
+# If you keep the line number information, uncomment this to
+# hide the original source file name.
+#-renamesourcefileattribute SourceFile
```
Comment (nicolbidone) [inline data/proguard-rules.pro]: missing empty line at end

File: di/.gitignore [hunk 5]
```diff
@@ -0,0 +1 @@
+/build
```
Comment (nicolbidone) [inline di/.gitignore]: missing empty line at end

File: di/build.gradle [hunk 6]
```diff
@@ -0,0 +1,45 @@
+plugins {
+    id 'com.android.library'
+    id 'kotlin-android'
+}
+
+android {
+    compileSdk 30
+
+    defaultConfig {
+        minSdk 26
```
Comment (nicolbidone) [inline di/build.gradle]: minSdk should be 23

File: di/proguard-rules.pro [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+#   public *;
+#}
+
+# Uncomment this to preserve the line number information for
+# debugging stack traces.
+#-keepattributes SourceFile,LineNumberTable
+
+# If you keep the line number information, uncomment this to
+# hide the original source file name.
+#-renamesourcefileattribute SourceFile
```
Comment (nicolbidone) [inline di/proguard-rules.pro]: missing empty line at end

File: domain/.gitignore [hunk 8]
```diff
@@ -0,0 +1 @@
+/build
```
Comment (nicolbidone) [inline domain/.gitignore]: missing empty line at end

**Review by nicolbidone — CHANGES_REQUESTED** [review]

File: app/build.gradle (hunk 1 above)
Comment (Faka-GLB) [reply app/build.gradle]: done

File: data/.gitignore (hunk 2 above)
Comment (Faka-GLB) [reply data/.gitignore]: added

File: data/build.gradle (hunk 3 above)
Comment (Faka-GLB) [reply data/build.gradle]: done

File: data/proguard-rules.pro (hunk 4 above)
Comment (Faka-GLB) [reply data/proguard-rules.pro]: added

File: di/.gitignore (hunk 5 above)
Comment (Faka-GLB) [reply di/.gitignore]: added

File: di/build.gradle (hunk 6 above)
Comment (Faka-GLB) [reply di/build.gradle]: done

File: domain/.gitignore (hunk 8 above)
Comment (Faka-GLB) [reply domain/.gitignore]: added

**Review by nicolbidone — APPROVED** [review]

**Pull request merged**